### PR TITLE
Manage go.mod and go.sum in Magic Modules

### PR DESCRIPTION
**Pull request by shuyama1**
This resolves https://github.com/hashicorp/terraform-provider-google/issues/8501

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/go.mod.erb
Comment (rileykarson) [inline mmv1/third_party/terraform/go.mod.erb:33]:
```suggestion
```

File: .ci/containers/terraform-tester/test_terraform.sh
Comment (rileykarson) [inline .ci/containers/terraform-tester/test_terraform.sh:29]: We can additionally remove the files from the exclusion list under https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/containers/downstream-builder/generate_downstream.sh#L95

**Review by rileykarson — COMMENTED** [review]
@ndmckinley was there an escape hatch built into the CI build killer to let us run through the whole change? (I'm assuming that was the cancellation source here)

File: mmv1/third_party/terraform/go.mod.erb
Comment (ndmckinley) [reply mmv1/third_party/terraform/go.mod.erb:33]: Yep!  You just add the PR number to the known-safe list on https://console.cloud.google.com/functions/details/us-central1/disallow-evil-builds?project=graphite-docker-images (whose name I can safely post here since it's IAM access controlled!)

I've done it just now for this PR.  Rerunning should work.

Comment (shuyama1) [reply mmv1/third_party/terraform/go.mod.erb:33]: Thanks!

File: mmv1/third_party/terraform/go.mod.erb
Comment (rileykarson) [inline mmv1/third_party/terraform/go.mod.erb:5]: nit: Let's tab-space the file instead of spaces, that's what's used on both downstreams

File: .ci/containers/terraform-tester/test_terraform.sh
Comment (rileykarson) [inline .ci/containers/terraform-tester/test_terraform.sh:32]: This change didn't appear to take! You'll probably have to build the image from your branch: https://github.com/GoogleCloudPlatform/magic-modules/tree/master/.ci/containers#updating-a-docker-image

File: mmv1/third_party/terraform/go.mod.erb
Comment (shuyama1) [reply mmv1/third_party/terraform/go.mod.erb:5]: Sure! Will work on it. Thanks!

File: .ci/containers/terraform-tester/test_terraform.sh
Comment (shuyama1) [reply .ci/containers/terraform-tester/test_terraform.sh:32]: Sure. Will do.

**Review by rileykarson — APPROVED** [review]

**Comment by rileykarson** [comment]
@shuyama1: Just curious, what are you looking for to change?

**Comment by shuyama1** [comment]
> @shuyama1: Just curious, what are you looking for to change by rerunning the build?

I did not make any changes but it seems that one check was in progress for a long time the last time it run. I thought I had t o have all the checks passed before merging it. Should I just merge instead?

**Comment by rileykarson** [comment]
VCR tests take an extreme amount of time, ~2 hours, and they're likely to fail for reasons unrelated to a given PR. If a PR hasn't modified a specific test case and I believe the change is low risk, I'll inspect the queue and merge if a couple tests have passed already. If a new test case was added I'll wait for it to pass, though, unless I run it manually in CI.

**Comment by shuyama1** [comment]
Sure. Thanks. Then I will just merge this PR as it seems fine last time I run the tests and nothing has been changed since then. Should I also rebase and rebuild the image for vcr-tester?

**Comment by shuyama1** [comment]
I think Stephen rebuilt the vcr-tester yesterday for CI Build fixes. What I mean is should I rebase his changes and rebuild it from my branch to also take my changes.

**Comment by rileykarson** [comment]
Oh, yup, in that case I'd rebase!

**Comment by shuyama1** [comment]
> Oh, yup, in that case I'd rebase!

Sure. Will do it. Thanks!

**Pull request merged**
